app/controllers/course/assessment/submissions_controller.rb

```diff
@@ -0,0 +1,41 @@
+# frozen_string_literal: true
+class Course::Assessment::SubmissionsController < Course::ComponentController
+  before_action :load_submissions
+  before_action :scope_submissions
+
+  def index #:nodoc:
+    @submissions = @submissions.
+                   includes(:assessment, experience_points_record: { course_user: :course }).
+                   page(page_param)
+  end
+
+  private
+
+  def submission_params
+    params.permit(:category)
+  end
+
+  # Load the current category, used to classify and load assessments.
+  def category
+    @category ||=
+      if submission_params[:category]
+        current_course.assessment_categories.find(submission_params[:category])
+      else
+        current_course.assessment_categories.first!
+      end
+  end
+
+  # Load the submissions based on the current category, and current role of the course_user.
+  def load_submissions
+    @submissions = Course::Assessment::Submission.
+                   where(assessment_id: category.assessments.select(:id)).
```

Comment:
make this a scope and test it ? then in controller test you can only test the if the correct category is loaded.